jvb/src/main/java/org/jitsi/videobridge/cc/allocation/BandwidthAllocator.java

```diff
@@ -548,17 +563,48 @@ public boolean hasNonZeroEffectiveConstraints(String endpointId)
         return sourceBitrateAllocations;
     }
 
+    /**
+     * Expire this bandwidth allocator.
+     */
+    void expire()
+    {
+        expired = true;
+    }
+
     /**
      * Submits a call to `update` in a CPU thread if bandwidth allocation has not been performed recently.
+     *
+     * Also, re-schedule the next update in at most {@code maxTimeBetweenCalculations}. This should only be run
+     * in the constructor or in the scheduler thread, otherwise it will schedule multiple tasks.
      */
-    void maybeUpdate()
+    private void rescheduleUpdate()
     {
-        if (Duration.between(lastUpdateTime, clock.instant())
-                .compareTo(BitrateControllerConfig.config.maxTimeBetweenCalculations()) > 0)
+        if (expired)
+        {
+            return;
+        }
+
+        Duration timeSinceLastUpdate = Duration.between(lastUpdateTime, clock.instant());
+        Duration period = BitrateControllerConfig.config.maxTimeBetweenCalculations();
+        long delayMs;
+        if (timeSinceLastUpdate.compareTo(period) > 0)
         {
-            logger.debug("Forcing an update");
+            logger.debug("Running periodic re-allocation.");
             TaskPools.CPU_POOL.execute(this::update);
+
+            delayMs = period.toMillis();
         }
+        else
+        {
+            delayMs = period.minus(timeSinceLastUpdate).toMillis();
+        }
+
+        // Add 5ms to avoid having to re-schedule right away. This increases the average period at which we
+        // re-allocate by an insignificant amount.
+        TaskPools.SCHEDULED_POOL.schedule(
```

Comment:
I'm wondering if we should hold on to the ScheduledFuture that this returns, so that `BandwidthAllocator.expire()` can cancel it.  Otherwise the bandwidth allocator object is retained by a reference from the scheduled executor service, which can be problematic if you're doing the "set it to something huge to disable it" hack (e.g. in your testing case where you set it to an hour).